src/number_format.js

```diff
@@ -644,11 +637,37 @@ class NumberFormat extends React.Component {
       return value;
     }
 
-    //if format got deleted reset the value to last value
-    if (this.checkIfFormatGotDeleted(start, end, lastValue)) {
-      value = lastValue;
+    // check whether the deleted portion has a character that is part of a format
+    const deletedValues = lastValue.substr(start, end - start);
+    const hasNonNumeric = !![...deletedValues].find((deletedVal, idx) => this.isCharacterAFormat(idx + start, lastValue));
+
+    // if it has, only remove characters that are not part of the format
+    if(hasNonNumeric) {
```

Comment:
hasNonNumeric variable name is confusing as format char can have number as well. Can we rename it to formatGotDeleted.